ignite/distributed/utils.py

```diff
@@ -46,19 +45,13 @@
 _need_to_sync = True
 
 
-def _sync_model_wrapper(func):
-    @wraps(func)
-    def wrapper(*args, **kwargs):
-        if isinstance(_model, _SerialModel) and _need_to_sync:
-            sync()
-        return func(*args, **kwargs)
-
-    return wrapper
-
-
-def sync():
+def sync(temporary=False):
```

Comment:
IMO `temporary` is obfuscating. Why something more explicit and related to the docstring, for instance `sync_every_call` ?

Reply:
Maybe, public API should not expose `temporary`. We can have 2 functions: `sync` with implicit `temporary=False` and `_sync_once` with implicit `temporary=True` and used inside `idist.get_*`

Reply:
But more complicated. Maybe have a unique function with secure mode by default should be enough.

Reply:
> Maybe have a unique function with secure mode by default should be enough.

What do you mean by secure mode  ?

Reply:
Secure mode means it always works, so always sync.

Reply:
@sdesrozis so are we good with current API or you propose to change it or its behaviour ?